connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufUtils.scala

```diff
@@ -178,46 +176,73 @@ private[sql] object ProtobufUtils extends Logging {
   }
 
   def buildDescriptor(descFilePath: String, messageName: String): Descriptor = {
-    val descriptor = parseFileDescriptor(descFilePath).getMessageTypes.asScala.find { desc =>
-      desc.getName == messageName || desc.getFullName == messageName
-    }
+    val fileDescriptor = parseFileDescriptorSet(descFilePath)
+      .find(!_.getMessageTypes.asScala.find(desc =>
```

Comment:
Oh, we need to find() invocations and check is made twice. How about this? [Optional to do, only a suggestion]

     // Find the first message descriptor that matches the name.
     val descriptorOpt =  parseFileDescriptorSet(descFilePath)
         .flatMap { fileDesc => 
                fileDesc.getMessageTypes.asScala.find { desc => 
                      desc.getName == messageName || desc.getFullName == messageName
                }
         }.headOption

      descriptoOpt match {
         case Some(d) => d
         case None => throw QueryCompilationErrors.unableToLocateProtobufMessageError(messageName)
     }

Reply:
fixed